src/components/Search/index.tsx

```diff
@@ -55,33 +55,45 @@ const handleSearchResult = async (
   setSearchLoading(inputElement, t)
 
   try {
-    const { data } = await explorerService.api.fetchSearchResult(addPrefixForHash(query))
+    const queryKey = addPrefixForHash(query)
+    const { data } = await explorerService.api.fetchSearchResult(queryKey)
     clearSearchInput(inputElement)
     setSearchValue('')
 
-    switch (data.type) {
-      case SearchResultType.Block:
-        history.push(`/block/${data.attributes.blockHash}`)
-        break
-
-      case SearchResultType.Transaction:
-        history.push(`/transaction/${data.attributes.transactionHash}`)
-        break
-
-      case SearchResultType.Address:
-        history.push(`/address/${data.attributes.addressHash}`)
-        break
-
-      case SearchResultType.LockHash:
-        history.push(`/address/${data.attributes.lockHash}`)
-        break
-
-      case SearchResultType.UDT:
-        history.push(`/sudt/${query}`)
-        break
-
-      default:
-        break
+    const isNormalSearchResult = (input: unknown): input is Response.Wrapper<unknown, any> =>
+      !!(input as Response.Wrapper<unknown, unknown>).type && !!(input as Response.Wrapper<unknown, unknown>).attributes
+    const isSearchTypeIdResult = (input: unknown): input is SearchTypeIdResult => !!(input as SearchTypeIdResult).args
+
+    if (isSearchTypeIdResult(data)) {
+      history.push(`/script/${queryKey}/type`)
```

Comment:
I'm not sure if I totally understand this function. In my mind, the `query key` would be `args` of a `type id script`, and if it matches a contract, it will be returned without an error.

The matched contract can be referenced by `type id script`, i.e.,

```yaml
code hash: 0x00000000000000000000000000000000000000000000000000545950455f4944
hash type: type
args: query key
```

Then the contract page should be 
```
/script/${Hash(0x00000000000000000000000000000000000000000000000000545950455f4944, type, query key)}/type
```
---

Currently, the query key is code hash or type hash, which has been supported.
<img width="580" alt="image" src="https://github.com/Magickbase/ckb-explorer-frontend/assets/7271329/fcd3d2cb-f699-4d75-bb69-744be0111e66">

Reply:
looks like that the backend only supports search by type hash:

https://github.com/nervosnetwork/ckb-explorer/pull/1530/files#diff-240d8fa62d7d98e879be4a2dc4a31b67265baee47f7f419a67226a483c71ebd7R58-R62

```
 def find_type_script_by_type_id
    type_script = TypeScript.find_by(script_hash: query_key)
    { data: { args: type_script.args } } if type_script.present?
 end
```

Maybe we should add search by `args` here? and return a `script hash` for frontend to redirect to. @rabbitz

Reply:
Yes, the feature is about `search args` here

> The feature request is about the **args**, when the **args** is being searched, and if a script referenced by it is found, the page should be navigated to the corresponding script page.

Ref: https://github.com/Magickbase/ckb-explorer-public-issues/issues/210

Reply:
Sorry, I'll fix it right away.

Reply:
updated: [ed484ab](https://github.com/Magickbase/ckb-explorer-frontend/pull/176/commits/ed484abc3395312ba364dc631220154ae92d4c6f)